### ISSUE TITLE
FSDP vs. MiCS

### ISSUE THREAD

**Issue by Shawnxys2016** [pytorch/pytorch#109180]
### 📚 The doc issue

I have read the documentation and paper about PyTorch FSDP (https://pytorch.org/blog/introducing-pytorch-fully-sharded-data-parallel-api/). I know there are two different type model sharding strategies: one is full sharding, and another is hybrid sharding. Also, i have read the MiCS in the DeepSpeed (paper: https://www.vldb.org/pvldb/vol16/p37-zhang.pdf code: https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/runtime/zero/mics.py). 
What is the difference between the PyTorch FSDP's Hybrid sharding and MiCS ? It seems that both of them first perform a reduce scatter of the gradients in the sharding Group and then the AllReduce operation in the replication group. The documentation makes me a bit confused.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @svekars @carljparker @fegin

**Comment by awgu** [comment]
Scale-aware partitioning is the same as FSDP's hybrid sharding (`ShardingStrategy.HYBRID_SHARD`).

**Comment by stu1130** [comment]
@awgu Is there any plan to support HYBRID_SHARD for more than 1 node like MiCS?

**Comment by awgu** [comment]
You can do it right now if you pass in a tuple of process groups as (shard PG, replicate PG) to the `process_group` argument, where you manually construct the shard PG to be >1 node.

https://github.com/pytorch/pytorch/blob/cbc6213f5d2a43bde7cf49cfec5047782824bd6b/torch/distributed/fsdp/fully_sharded_data_parallel.py#L246-L254

We are working on improving the utilities/UX for initializing the appropriate process groups via `init_device_mesh()` (cc: @wz337 ).

**Comment by wz337** [comment]
@Shawnxys2016 

You can find examples of how to use init_device_mesh() to assist HSDP initialization here. 
You can see two different initialization of FSDP line 74 vs line 83. The difference is that when using FSDP's SHARDED_STATE_DICT, if device_mesh is passed in, you would get DTensor state_dict. If process_group tuples are passed in, you would get ShardedTensor state_dict. Note that we are in the process of deprecating ShardedTensor, hence the two paths here. 
https://github.com/wz337/pytorch/blob/main/test/distributed/fsdp/test_hsdp_dtensor_state_dict.py#L70-L97 

Regardless, you can use init_device_mesh to assist HSDP initialization in both cases.
